### PR TITLE
*: release 0.50

### PR DESCRIPTION
**Pull request by paulfantom**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Releasing 0.50 version of prometheus-operator. 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
+## 0.50.0 / 2021-08-17
+
+* [CHANGE] Remove deprecated flags `--config-reloader-memory` and `--config-reloader-cpu` in favor of `--config-reloader-memory-limit`, `--config-reloader-memory-request`, `--config-reloader-cpu-limit`, and `--config-reloader-cpu-request`. #3884
+* [CHANGE] Remove use of Kubernetes API versions being removed in v1.22. #4171
+* [FEATURE] Added support for OAuth2 authentication in remote read and remote write configuration. #4113
+* [FEATURE] Added OAuth2 configuration for ServiceMonitor, PodMonitor and Probe. #4170
+* [FEATURE] Adds `prometheus_operator_spec_shards` metric for exposing the number of shards set on prometheus operator spec. #4173
```
Comment (simonpasquier) [inline CHANGELOG.md:7]:
```suggestion
* [FEATURE] Added `prometheus_operator_spec_shards` metric for exposing the number of shards set on prometheus operator spec. #4173
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+* [FEATURE] Support for `Authorization` section in various prometheus sections. #4180
+* [FEATURE] Validate prometheus rules when generating rule file content. #4184
+* [FEATURE] Support `label_limit`, `label_name_length_limit` and `label_value_length_limit` configuration fields at the Prometheus CRD level as well as support individual limits per ServiceMonitor, PodMonitor and Probe resources. #4195
+* [FEATURE] Add sample and target limits to Probe. #4207
+* [FEATURE] Adds `send_exemplars` field to the `remote_write` configuration in Prometheus. #4215 #4160
```
Comment (simonpasquier) [inline CHANGELOG.md:12]:
```suggestion
* [FEATURE] Added `send_exemplars` field to the `remote_write` configuration in Prometheus. #4215 #4160
```

**Review by simonpasquier — COMMENTED** [review]
we need to bump the version of the `github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring` module in `pkg/client/go.mod` too (it's missing from the RELEASE.md doc AFAICT).

**Comment by paulfantom** [comment]
> we need to bump the version of the github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring

bumped all versions and added instruction to RELEASE.md

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**